src/Displaz.jl

```diff
@@ -286,6 +288,10 @@ function plot3d(plotobj::DisplazWindow, position; color=[1,1,1], markersize=[0.1
     if size(color,2) == 1
         color = repmat(color, 1, nvertices)
     end
+
+    extra_fields = map(x -> (size(x[2]) == (nvertices,) || error("extra fields must be vectors of same length as number of points in position array") ;
+                                        (x[1], array_semantic, map(Float32, x[2]).')), kwargs) # works for vectors only at this stage...
```

Comment:
magic?

`kwargs` comes from Julia. It is a list of `Tuple{Symbol, Any}` as `(:keyword, transposed_vector)`.

I'm creating an anonymous function which changes the user's label `x[1]` and value `x[2]` into the format wanted by `write_ply_points()` which is `(:label, array_semantic, transposed_vector)`, but the anonymous function also verifies the number of elements in the transposed vector is correct.